contracts/token/ERC20/ERC20.sol

```diff
@@ -37,6 +37,52 @@ contract ERC20 is Context, IERC20 {
 
     uint256 private _totalSupply;
 
+    string private _name;
+    string private _symbol;
+    uint8 private _decimals;
+
+    /**
+     * @dev Sets the values for `name`, `symbol`, and `decimals`. All three of
+     * these values are immutable: they can only be set once during
+     * construction.
+     */
+    constructor (string memory name, string memory symbol, uint8 decimals) public {
```

Comment:
What do you think about hardcoding `decimals = 18`? There may be some interesting discussion around this in https://github.com/ethereum/EIPs/issues/724.

We could add a `_setDecimals` function as an escape hatch.

Reply:
We'd be opening the door to `decimals` changing after deployment, but I don't think I'll lose much sleep over that.

Reply:
Unless we make the function "constructor only"/